### PR TITLE
Custom dates

### PR DESCRIPTION
**Pull request by ayan-usgs**
Before making a pull request
----------------------------

- [x] Make sure all tests run
- [x] Update the changelog appropriately

Title
-----------
Brief description of changes. Reference the JIRA ticket if appropriate

Description
-----------
If no JIRA ticket is referenced, describe the changes made. Note anything that you want the reviewers to know while
reviewing your pull request

After making a pull request
---------------------------
- [x] If appropriate, put the link to the PR in the JIRA ticket
- [x] Assign someone to review unless the change is trivial

### CODE REVIEW

File: assets/src/scripts/selectors/time-series-selector.js [hunk 1]
```diff
@@ -122,9 +149,11 @@ export const getRequestTimeRange = memoize((tsKey, period, parmCd) => createSele
                 end: notes.requestDT
             };
         } else {
+            let intervalStart = notes['filter:timeRange'].interval.start;
```
Comment (mbucknell) [inline assets/src/scripts/selectors/time-series-selector.js]: I'd like to a see a test case in time-series-selector.spec.js for this additional functionality.

File: assets/src/scripts/store/index.js [hunk 2]
```diff
@@ -78,7 +79,7 @@ export const Actions = {
                     const notes = collection.queryInfo[requestKey].notes;
                     const endTime = notes.requestDT;
                     const startTime = calcStartTime('P7D', endTime, 'local');
-
+                    dispatch(Actions.setRequestedDates(startTime, endTime));
```
Comment (mbucknell) [inline assets/src/scripts/store/index.js]: I find this name confusing. Could we call it setCustomDateRange?

File: assets/src/scripts/store/index.js [hunk 3]
```diff
@@ -289,6 +325,23 @@ export const Actions = {
             period
         };
     },
+    setRequestedDates(startTime, endTime) {
+        return {
+            type: 'SET_REQUESTED_DATES',
```
Comment (mbucknell) [inline assets/src/scripts/store/index.js]: Ditto with the name here, maybe SET_CUSTOM_DATE_RANGE. Of course, I see this bubbles into other places but in my mine all of the date ranges are requested, so it's a little confusing more me. YMMV.

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ## [Unreleased]
 ### Changed
 - Graph server now serves routes from /api/graph-images
+- Added support for user specified date ranges
```
[the remark this answers is not in the document]
Comment (ayan-usgs) [reply CHANGELOG.md]: Ah, yes... :sweat_smile:

File: assets/src/scripts/components/hydrograph/axes.spec.js [hunk 5]
```diff
@@ -93,5 +118,47 @@ describe('Chart axes', () => {
...
+        it('custom uses monthly marks for 30 days through a year', () => {
+            const result = generateDateTicks(startP1Y, endDate, 'custom', timeZone);
+            expect(result.length).toBeGreaterThanOrEqual(6);
+            expect(result.includes(1475305200000)).toBe(true);
+            expect(result.includes(1480579200000)).toBe(true);
+            expect(result.includes(1496300400000)).toBe(true);
+            expect(result.includes(1501570800000)).toBe(true);
+        });
+
+        it('custom ticks are correctly generated for dates under one date', () => {
```
[the remark this answers is not in the document]
Comment (ayan-usgs) [reply assets/src/scripts/components/hydrograph/axes.spec.js]: One decade actually. :flushed:

File: assets/src/scripts/store/index.js (hunk 2 above)
Comment (ayan-usgs) [reply assets/src/scripts/store/index.js]: Sure.

File: assets/src/scripts/store/index.js (hunk 3 above)
Comment (ayan-usgs) [reply assets/src/scripts/store/index.js]: Done.

**Pull request merged**